test/gtest/cfg.cpp

```diff
@@ -290,3 +292,189 @@ TEST_F(CFGTest, FinitePowersetLatticeFunctioning) {
   element2.print(ss);
   EXPECT_EQ(ss.str(), "101101");
 }
+
+TEST_F(CFGTest, LinearReachingDefinitions) {
+  auto moduleText = R"wasm(
+    (module
+      (func $bar
+        (local $a (i32))
+        (local $b (i32))
+        (local $c (i32))
+        (local.set $a
+          (i32.const 1)
+        )
+        (drop
+          (local.get $a)
+        )
+        (local.set $b
+          (local.get $a)
+        )
+        (drop
+          (local.get $c)
+        )
+        (local.set $c
+          (i32.const 1)
+        )
+        (local.set $a
+          (i32.const 2)
+        )
+      )
+    )
+  )wasm";
+
+  Module wasm;
+  parseWast(wasm, moduleText);
+
+  Function* func = wasm.getFunction("bar");
+  CFG cfg = CFG::fromFunction(func);
+
+  LocalGraph::GetSetses getSetses;
+  LocalGraph::Locations locations;
+  FinitePowersetLattice<LocalSet*> lattice =
+    ReachingDefinitionsTransferFunction::getLatticeForFunc(func);
+  ReachingDefinitionsTransferFunction transferFunction(
+    lattice, func->getNumLocals(), getSetses, locations);
+
+  MonotoneCFGAnalyzer<FinitePowersetLattice<LocalSet*>,
+                      ReachingDefinitionsTransferFunction>
+    analyzer(lattice, transferFunction, cfg);
+  analyzer.evaluateFunctionEntry(func);
+  analyzer.evaluateAndCollectResults();
+
+  FindAll<LocalSet> foundSets(func->body);
+  FindAll<LocalGet> foundGets(func->body);
+  LocalGraph::GetSetses expectedResult;
+  expectedResult[foundGets.list[0]].insert(foundSets.list[0]);
+  expectedResult[foundGets.list[1]].insert(foundSets.list[0]);
+  expectedResult[foundGets.list[2]].insert(nullptr);
```

Comment:
It might be nice to use a list initializer for `expectedResult`, e.g.

```
LocalGraph::getSetses expectedResult = {{foundGets.list[0], {foundSets.list[0]}}, ...};
```

Further, it might be nice to give all the different gets and sets meaningful names (e.g. `getA`) to make it easier to match them up with the input program.

Reply:
List initializer doesn't seem to be supported for the `SmallSet`, but I will use more meaningful names.